src/plugins/lua/external_services.lua

```diff
@@ -181,12 +181,19 @@ local function add_scanner_rule(sym, opts)
       fun.each(function(p)
         local content = p:get_content()
         if content and #content > 0 then
-          cfg.check(task, content, p:get_digest(), rule)
+          local in_fname = p:get_filename()
+          local in_type, in_stype = p:get_detected_type()
+          local part_info = {in_fname,in_type,in_stype}
+          cfg.check(task, content, p:get_digest(), rule, part_info)
```

Comment:
I think the original idea was just to pass `mime_part` object (hence `p`) and not to split it into some array.

Reply:
You are absolutely right. How I couldn't think of that... I found that maybe_part is always nil, but it didn't occur to me to just pass p object instead of doing all these variable assignments here.
However, will that mean that the extraction of content type and subtype, as well as filename should go in icap.lua in lua_scanners?